### PR TITLE
Add local storage

### PR DESCRIPTION
**Pull request by joybiswas704**

### CODE REVIEW

File: src/components/Cart/Cart.js [hunk 1]
```diff
@@ -18,65 +18,66 @@ import OrderSummary from '../Checkout/OrderSummary';
...
+                            </VStack>
+                        </GridItem>
+                    </SimpleGrid>
+                </Flex>
+            </Elements>
+        </div>
+    );
 };
 
 export default Cart;
```
Comment (zealptl) [inline src/components/Cart/Cart.js:83]: This is the entire cart page right? Maybe it would be better to move to `src/pages`

Comment (joybiswas704) [reply src/components/Cart/Cart.js:83]: yes its the entire cart page. Ok i will move it.

File: src/components/Cart/CartItem.js [hunk 2]
```diff
@@ -1,85 +1,88 @@
...
+                src={imageURL}
+                fallbackSrc='https://via.placeholder.com/350'
+                alt={altImage}
+                borderRadius='16px'
+                boxSize='150px'
+                objectFit='contain'
+            />
+            <Box p={4}>
+                <Heading size='l'>{title}</Heading>
+                <Text maxW='400px'>{description}</Text>
```
Comment (zealptl) [inline src/components/Cart/CartItem.js:59]: add `noOfLines={3}` or `noOfLines={4}` prop to limit number of lines. whatever looks better

File: src/components/Cart/Cart.js [hunk 3]
```diff
@@ -18,65 +18,66 @@ import OrderSummary from '../Checkout/OrderSummary';
...
+                                    </Link>
+                                </Text>
+                            </VStack>
+                        </GridItem>
+                    </SimpleGrid>
+                </Flex>
+            </Elements>
+        </div>
+    );
 };
```
Comment (zealptl) [inline src/components/Cart/Cart.js:81]: very minor thing but can you add a bottom margin here so this component is not flush w the botto
Screen Shot 2021-12-04 at 12 00 00 AM
m

File: src/components/Cart/CartItem.js (hunk 2 above)
Comment (joybiswas704) [reply src/components/Cart/CartItem.js:59]: i will add noOfLines={4}. i didnt know about the prop. Thanks!

File: src/components/Cart/Cart.js (hunk 3 above)
Comment (joybiswas704) [reply src/components/Cart/Cart.js:81]: done

File: src/components/CartContextWrapper/CartContextWrapper.js [hunk 4]
```diff
@@ -1,98 +1,99 @@
...
-      console.log(error);
-    }
-  };
+    const calculateTotalItemsInCart = () => {
+        let totalItem = 0;
+        try {
+            cart.forEach((product) => {
+                totalItem = totalItem + parseInt(product.quantity);
+            });
+            return totalItem;
```
Comment (zealptl) [inline src/components/CartContextWrapper/CartContextWrapper.js:20]: if you want you can simply use reduce method like this

`return cart.reduce((accum, product) => accum + parseInt(product.quantity), 0)`

File: src/components/CartContextWrapper/CartContextWrapper.js [hunk 5]
```diff
@@ -1,98 +1,99 @@
...
-  };
+    const calculateTotalPriceInCart = () => {
+        let totalPrice = 0;
+        try {
+            cart.forEach((product) => {
+                totalPrice =
+                    totalPrice +
+                    product.quantity * parseFloat(product.unitPrice);
+            });
+            return totalPrice.toFixed(2);
```
Comment (zealptl) [inline src/components/CartContextWrapper/CartContextWrapper.js:34]: same thing here

`return cart
                 .reduce((accum, product) => 
                                     accum + parseInt(product.quantity) * parseFloat(product.unitPrice), 0)
                 .toFixed(2)`

File: src/components/Checkout/OrderSummary.js [hunk 6]
```diff
@@ -73,18 +83,43 @@ const OrderSummary = ({ checkout }) => {
...
+          });
           break;
-        case 'processing':
-          setMessage('Your payment is processing.');
+        }
+
+        case 'processing': {
+          toast({
+            title: 'Payment processing.',
+            status: 'warning',
```
Comment (zealptl) [inline src/components/Checkout/OrderSummary.js:99]: maybe change status to `info` instead of warning

File: src/components/Cart/CartItem.js [hunk 7]
```diff
@@ -1,85 +1,88 @@
...
+                <CloseButton
+                    onClick={() => handleRemoveItemFromCart(id)}
+                    m={2}
+                />
+            </Flex>
+        </Flex>
+    );
 };
 
 export default CartItem;
```
Comment (zealptl) [inline src/components/Cart/CartItem.js:88]: add unit test when you get a chance for this component

File: src/components/CartContextWrapper/CartContextWrapper.js (hunk 4 above)
Comment (joybiswas704) [reply src/components/CartContextWrapper/CartContextWrapper.js:20]: done

File: src/components/CartContextWrapper/CartContextWrapper.js (hunk 5 above)
Comment (joybiswas704) [reply src/components/CartContextWrapper/CartContextWrapper.js:34]: done

File: src/components/Checkout/OrderSummary.js (hunk 6 above)
Comment (joybiswas704) [reply src/components/Checkout/OrderSummary.js:99]: done

File: src/components/Checkout/OrderSummary.js [hunk 8]
```diff
@@ -1,178 +1,209 @@
...
+    }, [stripe]);
+
+    return (
+        <VStack
+            w='full'
+            p={10}
+            spacing={10}
+            alignItems='flex-start'
+            style={{
+                border: '2px solid #E2E8F0',
```
Comment (zealptl) [inline src/components/Checkout/OrderSummary.js:134]: sorry forgot can you change border to 1 px

File: src/pages/Cart.js [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+            <Elements stripe={stripePromise}>
+                <Flex h='100vh' py={20}>
+                    <SimpleGrid columns={3} rowGap={6} columnGap={3} w='full'>
+                        <GridItem colSpan={2}>
+                            <Box
+                                w='full'
+                                h='full'
+                                p={5}
+                                spacing={10}
+                                borderWidth='2px'
```
Comment (zealptl) [inline src/pages/Cart.js:40]: 1 px border

File: src/components/Checkout/OrderSummary.js (hunk 8 above)
Comment (joybiswas704) [reply src/components/Checkout/OrderSummary.js:134]: done

File: src/pages/Cart.js (hunk 9 above)
Comment (joybiswas704) [reply src/pages/Cart.js:40]: done

**Pull request merged**